controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -349,7 +351,15 @@ func (r *KubeadmControlPlaneReconciler) computeDesiredMachine(kcp *controlplanev
 	annotations := map[string]string{}
 	if existingMachine == nil {
 		// Creating a new machine
-		machineName = names.SimpleNameGenerator.GenerateName(kcp.Name + "-")
+		nameTemplate := "{{ .kubeadmcontrolplane.name }}-{{ .random }}"
+		if kcp.Spec.MachineNamingStrategy != nil && kcp.Spec.MachineNamingStrategy.Template != nil {
+			nameTemplate = *kcp.Spec.MachineNamingStrategy.Template
+		}
+		generatedMachineName, err := topologynames.KCPMachineNameGenerator(nameTemplate, kcp.Name).GenerateName()
```

Comment:
> If the templated string exceeds 63 characters, it will be trimmed to 58 characters and will
get concatenated with a random suffix of length 5.

should there be an explicit test case for this on the MachineNamingStrategy level?
presumably https://github.com/kubernetes-sigs/cluster-api/blob/90c4712f67bf21fce52e07a57488f62497306699/internal/topology/names/names.go#L40C30-L40C43 has tests, but might be worth testing on this level too.